.github/workflows/build-package-docs.yaml

```diff
@@ -120,3 +124,76 @@ jobs:
         name: package-docs-build
         path: build-directory/docs/docsite/ansible-package-docs-html-*.tar.gz
         retention-days: 7
+
+  deploy-package-docs:
+    needs: build-package-docs
+    runs-on: ubuntu-latest
+    environment:  # TODO create environment and specify it here
```

Comment:
@oraNod FYI environments are auto-created if you specify their names here. You can edit and configure them via the repo settings later on. No need to postpone filling this out. I recommend using the full format and specifying separate `name` and `url` fields here — it will show up on the UI nicely.

Reply:
this got me rethinking some things, particularly around the conditionals.

we should add an input so that the `deploy-package-docs` job runs only when specified. there might be times when we only want to do a build without pushing. without an `if:` statement for the job, it always fails unless you select the option to deploy to production. that seems less than ideal.

also, in future, we can add some steps to push to a "test" site, which will be a `gh-pages` branch in the destination repo. in this case it makes sense to use the same environment and most of the steps for pushing to production.